compiler/llvm/llvmDebug.cpp

```diff
@@ -384,23 +501,25 @@ llvm::DIType* debug_data::construct_type(Type *type) {
   } else if (ty->isStructTy() && type->astTag == E_AggregateType) {
     return construct_type_for_aggregate(
       llvm::cast<llvm::StructType>(ty), toAggregateType(type));
-  } else if (ty->isStructTy() && type->astTag == E_PrimitiveType &&
-             type->symbol->hasFlag(FLAG_EXTERN)) {
-    // Handle extern structs
+  } else if (ty->isStructTy() && type->astTag == E_PrimitiveType) {
+    // Handle extern and opaque structs as a forward decl
     llvm::StructType* struct_type = llvm::cast<llvm::StructType>(ty);
-
-    N = dibuilder->createForwardDecl(
-      llvm::dwarf::DW_TAG_structure_type,
-      name,
-      get_module_scope(defModule),
-      get_file(defModule, defFile),
-      defLine,
-      0, // RuntimeLang
-      !struct_type->isOpaque() ? layout.getTypeSizeInBits(ty) : 0,
-      !struct_type->isOpaque() ? 8*layout.getABITypeAlign(ty).value() : 0
-    );
-    type->symbol->llvmDIType = N;
-    return llvm::cast_or_null<llvm::DIType>(N);
+    if (type->symbol->hasFlag(FLAG_EXTERN) || struct_type->isOpaque()) {
+      N = dibuilder->createForwardDecl(
+        llvm::dwarf::DW_TAG_structure_type,
+        name,
+        get_module_scope(defModule),
+        get_file(defModule, defFile),
+        defLine,
+        0, // RuntimeLang
```

Comment:
I see this passed often enough - could we just create a magic constant for this in `llvmDebug.h` e.g., `constexpr int DEFAULT_RUNTIME_LANG = 0;`